File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res;
        int l = 0;
        int r = numbers.size() - 1;

        for (int i = 0; i < numbers.size(); i++){
            if(numbers[l] + numbers[r] > target){
                r -= 1;
            } else if(numbers[l] + numbers[r] < target){
                l += 1;
            } else{
                res.push_back(l+1);
                res.push_back(r+1);

                return res;
            }
        }
        
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevList = {}

        for i, n in enumerate(nums):
            diff = target - n #the difference b/w the target and the 
            # number, so we have to look if the number existes in teh preveMap

            if diff in prevList:
                return [prevList[diff], i]
            prevList[n] = i

        return  

        